<launch>

<arg name="map_path" default="$(find hdmap)/maps/"/>
<arg name="town" default="Town05"/>
<arg name="simulator" default="Carla"/>

<!-- 添加不确定性配置文件的路径参数 -->
<arg name="uncertainty_config" default="$(find planning_core)/config/perception_uncertainty.yaml"/>

<env name="LD_LIBRARY_PATH" value="$(env LD_LIBRARY_PATH):/$(find gp_planner)/thirdparty/gtsam-4.1rc/install/lib"/>
<node pkg="planning_core" name="planning" type="planning" output="screen">
  <param name="town" value="$(arg town)"/>
  <param name="simulator" value="$(arg simulator)"/>
  <param name="map_path" value="$(arg map_path)"/>
  <param name="random_drive_mode" value="true"/>

  <!-- 加载不确定性配置文件 -->
  <rosparam command="load" file="$(arg uncertainty_config)"/>
  
</node>

<node pkg="planning_core" name="carla_controller" type="carla_controller" >
  <param name="kp" value="3.5"/>
  <param name="ki" value="0.15"/>
  <param name="kd" value="0.4"/>
</node>

<!-- rviz -->
<!-- <arg name="rvizconfig" default="$(find planning_core)/rviz/planning.rviz"/>
<node name="planning_rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="false" /> -->

<!-- 3D model display -->
<arg name="model" default="$(find planning_core)/3d_model/3d_model.xacro" />
<param name="robot_description" command="$(find xacro)/xacro  $(arg model)" />
<node name="vehicle" pkg="robot_state_publisher" type="robot_state_publisher" />

</launch>